modules/mgidXBidAdapter.js

```diff
@@ -0,0 +1,263 @@
+    }
+    let location = refferLocation || winLocation;
+    const language = (navigator && navigator.language) ? navigator.language.split('-')[0] : '';
+    const host = location.host;
+    const page = location.pathname;
+    const secure = location.protocol === 'https:' ? 1 : 0;
+    const placements = [];
+    const request = {
+      deviceWidth,
+      deviceHeight,
+      language,
+      secure,
+      host,
+      page,
+      placements,
+      coppa: config.getConfig('coppa') === true ? 1 : 0,
+      ccpa: bidderRequest.uspConsent || undefined,
+      gdpr: bidderRequest.gdprConsent || undefined,
+      tmax: config.getConfig('bidderTimeout')
+    };
+
+    const len = validBidRequests.length;
+    for (let i = 0; i < len; i++) {
+      const bid = validBidRequests[i];
+      placements.push(getPlacementReqData(bid));
+    }
+
+    return {
+      method: 'POST',
+      url: AD_URL,
+      data: request
+    };
+  },
+
+  interpretResponse: (serverResponse) => {
+    let response = [];
+    for (let i = 0; i < serverResponse.body.length; i++) {
+      let resItem = serverResponse.body[i];
+      if (isBidResponseValid(resItem)) {
+        const advertiserDomains = resItem.adomain && resItem.adomain.length ? resItem.adomain : [];
+        resItem.meta = { ...resItem.meta, advertiserDomains };
+
+        response.push(resItem);
+      }
+    }
+    return response;
+  },
+
+  getUserSyncs: (syncOptions, serverResponses, gdprConsent, uspConsent, gppConsent) => {
+    const spb = isPlainObject(config.getConfig('userSync')) &&
+    isNumber(config.getConfig('userSync').syncsPerBidder)
+      ? config.getConfig('userSync').syncsPerBidder : USERSYNC_DEFAULT_CONFIG.syncsPerBidder;
+
+    if (spb > 0 && isPlainObject(syncOptions) && (syncOptions.iframeEnabled || syncOptions.pixelEnabled)) {
+      let pixels = [];
+      if (serverResponses &&
+        isArray(serverResponses) &&
+        serverResponses.length > 0 &&
+        isPlainObject(serverResponses[0].body) &&
+        isPlainObject(serverResponses[0].body.ext) &&
+        isArray(serverResponses[0].body.ext.cm) &&
+        serverResponses[0].body.ext.cm.length > 0) {
+        pixels = serverResponses[0].body.ext.cm;
+      }
+
+      const syncs = [];
+      const query = [];
+      query.push('cbuster={cbuster}');
+      query.push('gdpr_consent=' + encodeURIComponent(isPlainObject(gdprConsent) && isStr(gdprConsent?.consentString) ? gdprConsent.consentString : ''));
+      if (isPlainObject(gdprConsent) && typeof gdprConsent?.gdprApplies === 'boolean' && gdprConsent.gdprApplies) {
+        query.push('gdpr=1');
+      } else {
+        query.push('gdpr=0');
+      }
+      if (isPlainObject(uspConsent) && uspConsent?.consentString) {
+        query.push(`us_privacy=${encodeURIComponent(uspConsent?.consentString)}`);
+      }
+      if (isPlainObject(gppConsent) && gppConsent?.gppString) {
+        query.push(`gppString=${encodeURIComponent(gppConsent?.gppString)}`);
```

Comment:
what about sid

Reply:
sorry, didn't notice that. now we don't need sid and gppString params